llama-index-integrations/llms/llama-index-llms-deepinfra/llama_index/llms/deepinfra/base.py

```diff
@@ -0,0 +1,538 @@
+from llama_index.core.base.llms.generic_utils import get_from_param_or_env
+from llama_index.core.base.llms.types import (
+    CompletionResponse,
+    CompletionResponseGen,
+    ChatResponse,
+    ChatResponseGen,
+    ChatResponseAsyncGen,
+    CompletionResponseAsyncGen,
+    LLMMetadata,
+    MessageRole,
+    ChatMessage,
+)
+
+from llama_index.core.llms.callbacks import (
+    llm_chat_callback,
+    llm_completion_callback,
+)
+
+from llama_index.llms.deepinfra.utils import (
+    maybe_decode_sse_data,
+    chat_messages_to_list,
+    retry_request,
+    aretry_request,
+)
+
+"""DeepInfra API base URL."""
+API_BASE = "https://api.deepinfra.com/v1"
+"""DeepInfra Inference API endpoint."""
+INFERENCE_ENDPOINT = "inference"
+"""Chat API endpoint for DeepInfra."""
+CHAT_API_ENDPOINT = "openai/chat/completions"
+"""Environment variable name of DeepInfra API token."""
+ENV_VARIABLE = "DEEPINFRA_API_TOKEN"
+"""Default model name for DeepInfra embeddings."""
+DEFAULT_MODEL_NAME = "mistralai/Mixtral-8x22B-Instruct-v0.1"
+
+
+class DeepInfraLLM(LLM):
+    """DeepInfra LLM.
+
+    Examples:
+        `pip install llama-index-llms-deepinfra`
+
+        ```python
+        from llama_index.llms.deepinfra import DeepInfraLLM
+
+        llm = DeepInfraLLM(
+            model="mistralai/Mixtral-8x22B-Instruct-v0.1", # Default model name
+            api_key = "your-deepinfra-api-key",
+            temperature=0.5,
+            max_tokens=50,
+            additional_kwargs={"top_p": 0.9},
+        )
+
+        response = llm.complete("Hello World!")
+        print(response)
+        ```
+    """
+
+    model: str = Field(
+        default=DEFAULT_MODEL_NAME, description="The DeepInfra model to use."
+    )
+
+    temperature: float = Field(
+        default=DEFAULT_TEMPERATURE,
+        description="The temperature to use during generation.",
+        gte=0.0,
+        lte=1.0,
+    )
+    max_tokens: Optional[int] = Field(
+        description="The maximum number of tokens to generate.",
+        gt=0,
+    )
+
+    timeout: Optional[float] = Field(
+        default=None, description="The timeout to use in seconds.", gte=0
+    )
+    max_retries: int = Field(
+        default=10, description="The maximum number of API retries.", gte=0
+    )
+
+    _api_key: Optional[str] = PrivateAttr()
+
+    generate_kwargs: Dict[str, Any] = Field(
+        default_factory=dict, description="Additional keyword arguments for generation."
+    )
+
+    def __init__(
+        self,
+        model: str = DEFAULT_MODEL_NAME,
+        additional_kwargs: Optional[Dict[str, Any]] = None,
+        temperature: float = DEFAULT_TEMPERATURE,
+        max_tokens: Optional[int] = None,
+        max_retries: int = 10,
+        api_base: str = API_BASE,
+        timeout: Optional[float] = None,
+        api_key: Optional[str] = None,
+        callback_manager: Optional[CallbackManager] = None,
+        system_prompt: Optional[str] = None,
```

Comment:
I believe this is still not handled

Reply:
These parameters are used in parent and grandparent classes.